mailgun/management/commands/mailgun_delivery_times.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+import datetime
+import logging
+import requests
+from collections import defaultdict
+
+import pytz
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+
+DATE_FORMAT = '%a, %d %b %Y %H:%M:%S %z'
+EVENTS_MAX_PAGE_SIZE = 300
+EVENTS_OF_INTEREST = ('accepted', 'delivered')
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = ('Prints average and max delivery time for email sent in the past '
+            'N days.')
+
+    def add_arguments(self, parser):
+        parser.add_argument('--days-back', type=int, default=7,
+                            help='Number of days back to look at emails sent. '
+                                 '(default: %(default)s)')
+
+    def handle(self, *args, **options):
+        end = datetime.datetime.now(tz=pytz.utc)
+        begin = end - datetime.timedelta(days=options['days_back'])
+        logger.info('Pulling {} days of events for {}'.format(
+                        options['days_back'], settings.LISTSERV_DOMAIN))
+        events = self.get_events(begin, end)
+        logger.debug('Done pulling events')
+        self.process_events(events)
+
+    def get_events(self, begin, end):
+        # prime the pump
+        url = '{}{}/events'.format(settings.LISTSERV_API_URL,
+                                   settings.LISTSERV_DOMAIN)
+        auth = (settings.LISTSERV_API_USER, settings.LISTSERV_API_KEY)
+        params = {
+            'begin': begin.strftime(DATE_FORMAT),
+            'end': end.strftime(DATE_FORMAT),
+            'limit': EVENTS_MAX_PAGE_SIZE,
+            'event': ' OR '.join(EVENTS_OF_INTEREST),
+        }
+        resp = requests.get(url, auth=auth, params=params)
+        try:
+            resp.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            raise CommandError(str(e))
+        events = resp.json()['items']
+
+        # follow 'next' links until we get a blank page
+        while True:
+            url = resp.json()['paging']['next']
+            resp = requests.get(url, auth=auth)
+            try:
+                resp.raise_for_status()
```

Comment:
Why is the raise_for_status() done here instead of continuing on with getting the page(line 63) . I am trying to understand why we need this try block